### PR TITLE
Fixes to allow it to compile/generate FSM on 32/64 bits

### PR DESCRIPTION
**Pull request by mingodad**
Right now the generated FSM is not interchangeable between 32/64 bits

### CODE REVIEW

File: src/basil/bstream.lzz [hunk 1]
```diff
@@ -119,6 +128,146 @@ namespace basil
...
+  }
+
+  // read size_t
+  inline size_t readSize (TextReader & r)
+  {
+    size_t num;
+    r >> num;
+    return num;
+  }
+
```
Comment (mjspncr) [inline src/basil/bstream.lzz:270]: Is TextWriter just for debugging, looking at output?  Or is text meant to be an optional serialized format?

File: src/basilcc/serialize.lzz [hunk 2]
```diff
@@ -313,4 +328,6 @@ namespace
 #include <limits>
 #include <numeric>
 #include <stdexcept>
+//#define SerializeWriter BinaryWriter
+#define SerializeWriter TextWriter
```
Comment (mjspncr) [inline src/basilcc/serialize.lzz:332]: I see, so this is where you can switch to text. Is the idea to have a TextReader to deserialize the data?

Ideally you want to load the parser tables (the fsm) as fast as possible, so a binary format is used. But you are right that the binary format as it is here is not portable. So right now on a different architecture you'd have compile the lzz grammar (with basilcc) then use it with lzz. Really just need a binary format that *is* portable. Then you'd still have the speed.

File: src/basil/bstream.lzz (hunk 1 above)
Comment (mingodad) [reply src/basil/bstream.lzz:270]: It's an optional serializer that can be used for debug, and it's platform independent as you pointed out elsewhere, also the speed factor I'm not sure it'll a bottleneck and if it is we can have a cached binary version (read the text format and immediately save the platform specific binary version).

**Review by mjspncr — APPROVED** [review]
For debugging the text format would be helpful

**Pull request merged**